client/client.go

```diff
@@ -87,6 +87,24 @@ func (c *Client) doWithErr(req *http.Request) (*http.Response, error) {
 	return res, nil
 }
 
+func isAlphanumeric(word string) bool {
+	return regexp.MustCompile(`^[a-zA-Z0-9]*$`).MatchString(word)
+}
+
+func (c *Client) validateCredentials() error {
+	username, password := c.basicAuth()
+	err := &TwilioRestError{Status: 400}
+	if !isAlphanumeric(username) {
+		err.Message = "Invalid Username"
```

Comment:
This is going to be pretty vague and confusing for end users - if I get "Invalid Username" it might not be obvious to me where the problem is, especially if e.g. the user is pulling a username from an environment variable or a web form. 

Can we make the error clearer about what the problem is?